### PR TITLE
Skybox optimization

### PR DESCRIPTION
**Pull request by xubing0906**
https://github.com/cocos-creator/3d-tasks/issues/10482

### CODE REVIEW

File: cocos/core/renderer/scene/skybox.ts [hunk 1]
```diff
@@ -37,10 +37,38 @@ import { Root } from '../../root';
...
+     * @zh
+     * 半球反射和环境贴图反射
+     * @en
+     * hemisphere reflection and environment map reflection
+     * @readonly
+     */
+    AutoGen_Hemisphere_Diffuse_With_Reflection: 1,
+    /**
+     * @zh
+     * 漫反射贴图。
```
Comment (jk20012001) [inline cocos/core/renderer/scene/skybox.ts]: 说明改成 漫反射卷积图和环境反射

File: cocos/core/renderer/scene/skybox.ts [hunk 2]
```diff
@@ -37,10 +37,38 @@ import { Root } from '../../root';
...
+     * @zh
+     * 半球反射
+     * @en
+     * hemisphere reflection
+     * @readonly
+     */
+    Hemisphere_Diffuse: 0,
+    /**
+     * @zh
+     * 半球反射和环境贴图反射
```
Comment (jk20012001) [inline cocos/core/renderer/scene/skybox.ts]: 改成半球漫反射和环境反射

File: cocos/core/scene-graph/scene-globals.ts [hunk 3]
```diff
@@ -221,6 +221,8 @@ legacyCC.AmbientInfo = AmbientInfo;
 @ccclass('cc.SkyboxInfo')
 @help('i18n:cc.Skybox')
 export class SkyboxInfo {
+    @serializable
+    protected _environmentLightingType = EnvironmentLightingType.Hemisphere_Diffuse;
     @serializable
     protected _applyDiffuseMap = false;
```
Comment (jk20012001) [inline cocos/core/scene-graph/scene-globals.ts]: _applyDiffuseMap和_useIBL删掉, 只保留Type

**Review by jk20012001 — APPROVED** [review]

File: cocos/core/renderer/scene/skybox.ts [hunk 4]
```diff
@@ -37,10 +37,38 @@ import { Root } from '../../root';
...
 
+export const EnvironmentLightingType = Enum({
+    /**
+     * @zh
+     * 半球漫反射
+     * @en
+     * hemisphere diffuse
+     * @readonly
+     */
+    Hemisphere_Diffuse: 0,
```
Comment (pandamicro) [inline cocos/core/renderer/scene/skybox.ts]: ALL_CAPS for enum naming

File: cocos/core/scene-graph/scene-globals.ts [hunk 5]
```diff
@@ -222,7 +222,7 @@ legacyCC.AmbientInfo = AmbientInfo;
 @help('i18n:cc.Skybox')
 export class SkyboxInfo {
     @serializable
-    protected _applyDiffuseMap = false;
+    protected _environmentLightingType = EnvironmentLightingType.Hemisphere_Diffuse;
```
Comment (pandamicro) [inline cocos/core/scene-graph/scene-globals.ts]: Private or protected property can use shorter names like

`_envLightingType`

B.t.w, public properties `envmap` and `environmentLightingType` doesn't follow the same naming rule, I suggest to change them to : 

- `envMap`
- `envLightingType`

**Review by pandamicro — APPROVED** [review]

**Pull request merged**